### ISSUE TITLE
Windows error when creating new project: "Uncaught Error: ENOTEMPTY, directory not empty 'c:\Users\TTaylor\Documents\mmworkspace\Trailhead Org\unpackaged\staticresources'"

### ISSUE THREAD

**Issue by tet3** [joeferraro/MavensMate-Atom#125]
Win 7 x64, Atom 0.165.0, MM-Atom 0.0.14

This is a new project, so the project directory doesn't exist when starting the project creation. There is a single ZIP file in Static Resources. 

Removing StaticResource from the Mm default subscription doesn't seem to change this behavior, neither does removing .resource from the Mm apex file extensions setting. 

Once I close the error message, there is a project directory, with an unpackaged and unpackaged\staticresources folder, that has a .resource file in it. There is no config or src directory. 

Additionally, the "report an issue on the MavensMate-Atom project" button at the bottom of the error message opens a git.io URL that generates a 404.

Error message with stack trace:
```
Uncaught Error: ENOTEMPTY, directory not empty 'c:\Users\TTaylor\Documents\mmworkspace\Trailhead Org\unpackaged\staticresources'
c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\node_modules\asap\asap.js:45

Error: ENOTEMPTY, directory not empty 'c:\Users\TTaylor\Documents\mmworkspace\Trailhead Org\unpackaged\staticresources'
  at Error (native)
  at Object.fs.rmdirSync (fs.js:649:18)
  at rmkidsSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:247:11)
  at rmdirSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:237:7)
  at fixWinEPERMSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:150:5)
  at rimrafSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:216:26)
  at c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:245:5
  at Array.forEach (native)
  at rmkidsSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:244:26)
  at rmdirSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:237:7)
  at fixWinEPERMSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:150:5)
  at rimrafSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:216:26)
  at c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:245:5
  at Array.forEach (native)
  at rmkidsSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:244:26)
  at rmdirSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:237:7)
  at fixWinEPERMSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:150:5)
  at Function.rimrafSync (c:\Users\TTaylor\AppData\Local\atom\app-0.165.0\resources\app\node_modules\fs-plus\node_modules\rimraf\rimraf.js:216:26)
  at Object.rmrfSync (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\fs-extra\lib\remove.js:7:19)
  at c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\lib\mavensmate\project.js:304:14
  at Promise_then_rejected (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\q.js:780:43)
  at Promise_done_rejected (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\q.js:862:26)
  at Rejected_then [as then] (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\q.js:1324:23)
  at Rejected_dispatch [as dispatch] (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\q.js:1312:23)
  at Pending_become_eachMessage_task (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\q.js:1370:30)
  at flush (c:\Users\TTaylor\.atom\packages\MavensMate-Atom\node_modules\mavensmate\node_modules\q\node_modules\asap\asap.js:27:13)
  at process._tickDomainCallback (node.js:409:11)
```

**Comment by mattisdada** [comment]
Just to add some more details to this:

It also happens on Windows 8.1 and what is unusall is it creates a file in unpackaged/staticresources called "**SiteSamples.resource**" and it has 0kb, no permissions, no owner, it cannot be deleted or modified in anyway(seriously, I ran del as SYSTEM and it still errors) and once the system restarts it disappears entirely. 

I've only ever tried a Sandbox instance.

But I did see the project has had a new commit in the last 40mins, so this problem may already be resolved

**Comment by joeferraro** [comment]
thanks for the details guys. unfortunately, complex zip files aren't particularly well supported in node, but I'm determined to figure it out. Keep an eye out for more info on this issue ...

**Comment by itkevin** [comment]
I found a workaround for this so that you can at least create the Project:

1. Go to the Mavensmate Settings and change the "Mm default subscription" to `ApexTrigger`
2. Restart Atom and go to MavensMate -> Project -> New Project...
3. Enter all your details
4. Go to the Project Metadata Tab and select all or just a few in ApexTrigger
5. Create the project

After that it should create the project without the error and you can add the missing Metadata in Edit Project...
Hope this works for you as well.
I did this on 0.0.14

**Comment by joeferraro** [comment]
Hey guys, thanks for testing. Right now there is a potential Atom bug in Windows that could be causing these issues. https://github.com/atom/atom/issues/5481

**Comment by mattisdada** [comment]
In reply to the Atom thread, (I don't really have enough confidence in this to post it on the Atom official).

Have you tried testing it with Io.js version 1.2.0 instead of Node? It might not be a specific issue with Atom but Io.js instead (they switched from node.js to Io.js a couple of builds ago)

**Comment by indyHarcourt** [comment]
Any updates to this issue?  Is it being actively researched, or are we Windows users just out of luck?

**Comment by mattisdada** [comment]
@indyHarcourt  Mavensmate team is waiting for the issue atom/atom#5481 to be resolved, at the moment it is a roadblock. This week sometime I might investigate the issue and have a crack at it, but I am very doubtful I'll get anywhere with it....

**Comment by joeferraro** [comment]
It's likely this issue will force us to rewrite command executions as node subprocesses in a circumvention attempt. It's probably better from an Atom performance and memory management standpoint anyway, but it will impact the performance of MavensMate command calls, as the cli, in its current form, needs to bootstrap the project and client for each request.

**Comment by force2b** [comment]
Thanks for the update @joeferraro . Atom+MavensMate looks really powerful. Very much looking forward to eventually running it on my Windows box.

**Comment by venkatpolisetti** [comment]
I am getting the same error on Windows and could not proceed to create a project.

I have tried @itkevin method but it still gave me the same error. Hope this gets fixed soon so that I can go ahead use atom.io for SF Development.

**Comment by GradyD** [comment]
I just tried @itkevin's workaround on windows 8.1 using atom 201 and it worked like a charm!

**Comment by venkatpolisetti** [comment]
I had tried it on Windows 7 and did not work.

**Comment by GradyD** [comment]
@venkatpolisetti are you actually selecting any triggers? You can't leave the box unchecked.

**Comment by venkatpolisetti** [comment]
@GradyD Just checked it in a fresh install of Atom and Mavensmate-Atom on Windows 7 and followed @itkevin advice verbatim and ended up getting the same error.

I believe, Windows 8.1 may have a different security model than in Win 7.

Hope this gets fixed soon.

Thanks.

**Comment by joeferraro** [comment]
this should be resolved in v1.0.0. please feel free to re-open if you're still experiencing this issue. thanks!
